src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Platform/LoggerBase.cs

```diff
@@ -27,18 +27,52 @@
 
 using System;
 using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Text;
 
 namespace Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Platform
 {
     internal abstract class LoggerBase
     {
-        internal abstract void Verbose(CallState callState, string message, [System.Runtime.CompilerServices.CallerFilePath] string callerFilePath = "");
+        internal abstract void DefaultLog(LogLevel logLevel, string message);
 
-        internal abstract void Information(CallState callState, string message, [System.Runtime.CompilerServices.CallerFilePath] string callerFilePath = "");
+        private void Log(CallState callState, LogLevel logLevel, string message, bool containsPii,
+            [CallerFilePath] string callerFilePath = "")
+        {
+            if (!LoggerCallbackHandler.PiiLoggingEnabled && containsPii)
```

Comment:
you already have this method that does the check. you should not need if conditions above.

Reply:
changed